### PR TITLE
[chip-test] Fix flash_ctrl alert on FPGA target

### PR DESCRIPTION
**Pull request by moidx**
Update the alert_handler_lpg_sleep_mode_pings_test target to initialize the Creator and Owner info flash pages to avoid fatal alerts in the flash_ctrl. This change is only required on the FPGA target.

Fixes https://github.com/lowRISC/opentitan/issues/16356
Fixes https://github.com/lowRISC/opentitan/issues/16576

### CODE REVIEW

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 1]
```diff
@@ -401,16 +396,14 @@ static void execute_test_phases(uint8_t test_phase, uint32_t ping_timeout_cyc) {
...
-     * (ping_timeout_cnt = 256). End the test if any alerts or loc_alerts is
-     * fired.
-     *
-     * Phase #2: N/A (As the ping_timeout_cnt = 2 in this phase, the ping-fail
-     * alerts will come continuosly)
-     */
+    // Check if sleep/wakeup cycle caused any spurious alerts.
+    //
+    // Phase#1: No alerts or loc_alerts shall be fired
+    // (ping_timeout_cnt = 256). End the test if any alerts or loc_alerts is
```
Comment (engdoreis) [inline sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:402]: Nit. 
```suggestion
    // `(ping_timeout_cnt = 256)`. End the test if any alerts or loc_alerts is
```

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 2]
```diff
@@ -207,12 +208,6 @@ static uint16_t alert_handler_num_fired_alerts(void) {
   uint16_t accumulator = 0;
   // Loop over all alert_cause regs
   for (int jj = 0; jj < ALERT_HANDLER_PARAM_N_ALERTS; jj++) {
```
Comment (engdoreis) [inline sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:210]: Nit. 
```suggestion
  for (size_t alert = 0; alert < ALERT_HANDLER_PARAM_N_ALERTS; ++alert) {
```

**Review by engdoreis — APPROVED** [review]
Thanks @moidx  for the PR, it LGTM. I pointed out some minor changes, but feel free to ignore it.

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c (hunk 1 above)
Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c]: Done

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 3]
```diff
@@ -401,16 +396,14 @@ static void execute_test_phases(uint8_t test_phase, uint32_t ping_timeout_cyc) {
...
-     * Phase #2: N/A (As the ping_timeout_cnt = 2 in this phase, the ping-fail
-     * alerts will come continuosly)
-     */
+    // Check if sleep/wakeup cycle caused any spurious alerts.
+    //
+    // Phase#1: No alerts or loc_alerts shall be fired
+    // (ping_timeout_cnt = 256). End the test if any alerts or loc_alerts is
+    // fired.
+    //
+    // Phase #2: N/A (As the ping_timeout_cnt = 2 in this phase, the ping-fail
```
[the remark this answers is not in the document]
Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c]: Done

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 4]
```diff
@@ -420,16 +413,14 @@ static void execute_test_phases(uint8_t test_phase, uint32_t ping_timeout_cyc) {
...
-     * Phase #2: kDifAlertHandlerLocalAlertAlertPingFail shall be fired
-     * Phase #2: kDifAlertHandlerLocalAlertEscalationPingFail shall be fired.
-     */
+    // Check that ping mechanism still works as expected after sleep/wakeup
+    // cycle
+    //
+    // Phase #1: No alerts or loc_alerts shall be fired
+    //
+    // Phase #2: new alerts shall be fired
+    // Phase #2: kDifAlertHandlerLocalAlertAlertPingFail shall be fired
```
[the remark this answers is not in the document]
Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c]: Done

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c (hunk 2 above)
Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c]: Done

**Comment by msfschaffner** [comment]
Great, thanks for fixing this @moidx.

Does this also address https://github.com/lowRISC/opentitan/issues/16576 and https://github.com/lowRISC/opentitan/issues/16822?

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c [hunk 5]
```diff
@@ -510,6 +509,18 @@ bool test_main(void) {
...
 
+  dif_rstmgr_reset_info_bitfield_t rst_info = rstmgr_testutils_reason_get();
+  rstmgr_testutils_reason_clear();
+
+  // We need to initialize the info FLASH partitions storing the Creator and
+  // Owner secrets to avoid getting the flash controller into a fatal error
+  // state.
+  if (kDeviceType == kDeviceFpgaCw310 && rst_info & kDifRstmgrResetInfoPor) {
+    CHECK_STATUS_OK(keymgr_testutils_flash_init(&flash_ctrl, &kCreatorSecret,
+                                                &kOwnerSecret));
```
Comment (msfschaffner) [inline sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:520]: Is this something that needs to be added to other testcases, too? E.g. https://github.com/lowRISC/opentitan/issues/16822?

**Review by msfschaffner — APPROVED** [review]

File: sw/device/tests/alert_handler_lpg_sleep_mode_pings.c (hunk 5 above)
Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:520]: I will take a look.

Comment (moidx) [reply sw/device/tests/alert_handler_lpg_sleep_mode_pings.c:520]: Added one more commit to resolve https://github.com/lowRISC/opentitan/issues/16576, and marked https://github.com/lowRISC/opentitan/issues/16822 as obsolete.

**Pull request merged**